diff --git a/CMakeLists.txt b/CMakeLists.txt
index 98aeb3c..ad90da3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -78,4 +78,13 @@ install(TARGETS ticpp_ticpp
     DESTINATION ${CMAKE_INSTALL_LIBDIR}
     ${excludeFromAllTag}
 )
+
+  install(
+    DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/
+    DESTINATION ${CMAKE_INSTALL_PREFIX}/include/ticpp
+    FILES_MATCHING PATTERN "*.h"
+    PATTERN "*.git" EXCLUDE
+    PATTERN "build" EXCLUDE
+    PATTERN "tests" EXCLUDE
+    )
 unset(excludeFromAllTag)
